apps/test/unit/util/imageUtilsTest.js

```diff
@@ -16,12 +15,38 @@ const TEST_DATA_URI =
 describe('image utils', () => {
   it('overlays an image inside the Artist frame', done => {
     dataURIToFramedBlob(TEST_DATA_URI, actual => {
-      dataURIFromURI(expectedPng).then(expected => {
-        blobToDataURI(actual, actual => {
-          expect(expected).to.equal(actual);
-          done();
-        });
-      });
+      assertVisualMatch(expectedPng, actual)
+        .then(done)
+        .catch(done);
```

Comment:
it looks like this line will cause the unit test to pass if there is an assertion error within assertVisualMatch. removing `.catch(done)` seems like it would do what you want, and cause the test to fail if the match fails. Unless I am misunderstanding something here?

Reply:
The `done` callback doesn't necessarily mean success.

> This callback accepts both an `Error` instance (or subclass thereof) or a falsy value; anything else is invalid usage and throws an error (usually causing a failed test).
> [MochaJS.org / Asynchronous code](https://mochajs.org/#asynchronous-code)

The docs give this example code:

```js
user.save(function(err) {
  if (err) done(err);
  else done();
});
```

Since `catch` passes its callback an error object (in our case probably an assertion error), using `done` as the catch-callback will fail the test immediately with useful output instead of masking the assertion error with an unhandled promise rejection.

Reply:
good to know, thanks for explaining!